protocols/request-response/src/lib.rs

```diff
@@ -448,8 +448,11 @@ where
     /// by [`NetworkBehaviour::handle_pending_outbound_connection`].
     ///
     /// Addresses added in this way are only removed by `remove_address`.
-    pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) {
-        self.addresses.entry(*peer).or_default().push(address);
+    ///
+    /// Returns true if the address was added, false otherwise (i.e. if the
+    /// address is already in the list).
+    pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) -> bool {
```

Comment:
Ah, hang on, this is a breaking change. Mind making it a non-breaking change?


```suggestion
    ///
    /// Returns true if the address was added, false otherwise (i.e. if the
    /// address is already in the list).
    pub fn add_address(&mut self, peer: &PeerId, address: Multiaddr) {
```

Reply:
Good point, committed!